support/reftest/reftest.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+"""
+
+Prepare reference test data for Exodus-lambda integration test.
+
+"""
+
+import os
+import sys
+import yaml
+import boto3
+
+import hashlib
+import requests
+import argparse
+
+from tqdm import tqdm
+from datetime import datetime, timezone
+
+
+class DBHandler:
+    def __init__(
+        self,
+        table,
+        access_id=None,
+        access_key=None,
+        session_token=None,
+        region_name=None,
+    ):
+        self.dynamodb = boto3.Session(
+            aws_access_key_id=access_id,
+            aws_secret_access_key=access_key,
+            aws_session_token=session_token,
+            region_name=region_name,
+        ).client("dynamodb")
+
+        self.table = table
+
+    def search_item_from_web_uri(self, web_uri):
+        return self.dynamodb.query(
+            TableName=self.table,
+            Limit=1,
+            ScanIndexForward=False,
+            KeyConditionExpression="web_uri = :u and from_date <= :d",
+            ExpressionAttributeValues={
+                ":u": {"S": web_uri},
+                ":d": {
+                    "S": str(
+                        datetime.now(timezone.utc).isoformat(timespec="milliseconds")
+                    )
+                },
+            },
+        )
+
+    def search_object_key_from_web_uri(self, web_uri):
+        res = self.search_item_from_web_uri(web_uri)
+        if not res["Items"]:
+            return False
+        else:
+            return res["Items"][0]["object_key"]["S"]
+
+    def delete_item_from_web_uri(self, web_uri):
+        res = self.search_item_from_web_uri(web_uri)
+        return self.dynamodb.delete_item(
+            TableName=self.table,
+            Key={
+                "web_uri": {"S": res["Items"][0]["web_uri"]["S"]},
+                "from_date": {"S": res["Items"][0]["from_date"]["S"]},
+            },
+        )
+
+    def put_item(self, web_uri, object_key, from_date=None, metadata={}):
+        if not from_date:
+            from_date = datetime.utcnow().isoformat(timespec="seconds")
```

Comment:
Why does this one use timespec="seconds" while the searching code uses timespec="milliseconds" ?